wdns/rr_to_ubuf.c

```diff
@@ -25,12 +25,19 @@ _wdns_rr_to_ubuf(ubuf *u, wdns_rr_t *rr, unsigned sec)
 	dns_type = wdns_rrtype_to_str(rr->rrtype);
 
 	if (sec == WDNS_MSG_SEC_QUESTION)
-		ubuf_add_cstr(u, ";");
+		ubuf_append_cstr_lit(u, ";");
 
 	ubuf_add_cstr(u, name);
 
-	if (sec != WDNS_MSG_SEC_QUESTION)
-		ubuf_add_fmt(u, " %u", rr->rrttl);
+	if (sec != WDNS_MSG_SEC_QUESTION) {
+		char tmp[10];
```

Comment:
this is why sometimes just padding to a power of 2 (16) is nice for multiple reasons.
rrttl is type uint32_t so the largest possible value is 4294967295 - which is 10 characters long PLUS the trailing nul byte.... so an off by one overflow is possible here.